source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -291,4 +319,22 @@ namespace PlayFab
             return activeRequestCount;
         }
     }
+
+    void PlayFabCurlHttpPlugin::CurlHeaderFailed(CallRequestContainer& requestContainer)
+    {
+        requestContainer.errorWrapper.HttpStatus = "Failed to create Headers list";
+        requestContainer.errorWrapper.ErrorCode = PlayFabErrorCode::PlayFabErrorUnkownError;
+        requestContainer.errorWrapper.ErrorName = "Header Creation failed";
+        requestContainer.errorWrapper.ErrorMessage = "Request failed initializing before sending. Failing out early.";
+    }
+
+    curl_slist* PlayFabCurlHttpPlugin::TryCurlAddHeader(CallRequestContainer& requestContainer, curl_slist* list, const char* headerToAppend)
+    {
+        list = curl_slist_append(list, headerToAppend);
```

Comment:
```suggestion
        if (list != nullptr)
        {
            list = curl_slist_append(list, headerToAppend);
        }
```

Reply:
list starts as a nullptr. This code prevents us from ever adding to the curl_slist (see line 165, it's the most commonly suggested way to be calling curl_slist_append by passing in the nullptr first https://curl.haxx.se/libcurl/c/curl_slist_append.html

This entire PR is only to check for the pointer to the list coming out. if that is a nullptr, something went wrong, and we need to end whatever we are doing.